lxd/storage/drivers/driver_zfs.go

```diff
@@ -365,14 +365,11 @@ func (d *zfs) Validate(config map[string]string) error {
 
 			return validate.IsBool(value)
 		}),
-		"zfs.export":                  validate.Optional(validate.IsBool),
-		"volume.zfs.blocksize":        validate.Optional(ValidateZfsBlocksize),
-		"volume.zfs.remove_snapshots": validate.Optional(validate.IsBool),
-		"volume.zfs.use_refquota":     validate.Optional(validate.IsBool),
-		"volume.zfs.reserve_space":    validate.Optional(validate.IsBool),
+		"zfs.export":           validate.Optional(validate.IsBool),
+		"volume.zfs.blocksize": validate.Optional(ValidateZfsBlocksize),
```

Comment:
Why isnt "volume.zfs.blocksize" covered by ZFS' commonVolumeRules?

Reply:
This is the same case as below so I left this as is was.

Reply:
I think this should be valid as `zfs.blocksize` at the volume level too.

Reply:
And this is valid on volume level but I did't put this in `commonVolumeRules` as on volume level there is different validator so even if I put this to `commonVolumeRules` then It will require overriding on pool level.

Reply:
No, I think we should have `"zfs.blocksize": validate.Optional(ValidateZfsBlocksize),` in the `commonVolumeRules` list (which can then be passed into d.validatePool()` and then for `ValidateVolume()` below we should do something like this:

```go
// ValidateVolume validates the supplied volume config.
func (d *zfs) ValidateVolume(vol Volume, removeUnknownKeys bool) error {
	rules := d.commonVolumeRules()
	rules["zfs.blocksize"] = validate.Optional(func(value string) error {
		if vol.contentType != ContentTypeFS {
			return fmt.Errorf("Blocksize can be change only for filesystem type")
		}

		return rules["zfs.blocksize"](value) // Use the common validation after checking volume content type.
	})

	return d.validateVolume(vol, rules, removeUnknownKeys)
}
```

This way we avoid having to make exceptions in the common rules, and stack common validators with volume specific validators at the volume level. This way if the common validator changes in the future, we don't have to change it in two places.

Reply:
This approach should work to get the value of `rules["zfs.blocksize"]` from `d.commonVolumeRules()` before we replace it. But if not, then loading its initial value into a temporary variable and using that inside the custom validator would work.

Reply:
Done. Temporary variable was needed.